homeassistant/components/device_automation/trigger.py

```diff
@@ -20,28 +24,12 @@
 TRIGGER_SCHEMA = DEVICE_TRIGGER_BASE_SCHEMA.extend({}, extra=vol.ALLOW_EXTRA)
 
 
-class DeviceAutomationTriggerProtocol(Protocol):
+class DeviceAutomationTriggerProtocol(TriggerProtocol, Protocol):
```

Comment:
Does this need double inheritance?

Reply:
Without inheriting Protocol again,  the derived class is an implementation of the Protocol with some extra methods: 
https://mypy.readthedocs.io/en/stable/protocols.html#defining-subprotocols-and-subclassing-protocols

Reply:
Cancel that - it is needed for sub-protocols
https://mypy.readthedocs.io/en/stable/protocols.html#defining-subprotocols-and-subclassing-protocols